docs/adr/0009_versioning_strategy.md

```diff
@@ -0,0 +1,305 @@
+* **Source determination:** Create branch from `main` for new minor version release
+* **Branch creation:** Create `releases/<component>/v<major>.<minor>` branch using OCM bot
+* **VERSION file initialization:** Ensure component's `VERSION` file reflects the target minor version
+* **Protection rules:** Apply branch protection (require PR reviews, status checks)
+* **Notification:** Alert component maintainers of new release branch creation
+
+**Manual Parameters:**
+
+* `component`: Target component name (e.g., `ocm-cli`, `ocm-controller`)
+* `major.minor`: Target version (e.g., `0.30` for `releases/ocm-cli/v0.30`)
+* `source-ref`: Source branch/tag to create from (default: `main`)
+* `initial-patch`: Starting patch version (default: `0`)
+
+**Permissions Required:** OCM Bot token with branch creation and protection rule management
+
+**Validation Checks:**
+
+* Component exists in repository structure
+* Source reference is valid and accessible
+* Version format follows SemVer conventions
+* No conflicting release branch already exists
+
+**Error Handling:**
+
+* Duplicate branch: Provide guidance on using existing branch or incrementing minor version
+* Invalid component: List available components and naming conventions
+* Permission failures: Alert repository administrators
+
+**Note:** OCM root component does not require release branches - it uses tags directly from `main` branch.
+
+#### Workflow: `release-from-branch`
+
+**Trigger:** `push` to `releases/<component>/**` OR manual workflow dispatch
+
+**Required Steps:**
+
+* Pre-flight checks: validate branch naming convention and component existence
+* Path filtering: only trigger if changes detected in component directory or VERSION file
+* Run component-specific tests and verification
+* Read `cli/VERSION` or `kubernetes/controller/VERSION` (fall back to latest component tag if absent)
+* Compute new patch number using release tooling
+* Create annotated tag `<component>/vX.Y.Z` using OCM bot
+* Build and publish artifacts (binaries, container images, component descriptors)
+* **Root component handling (configurable):**
+  * **Default:** Calculate root component SemVer bump and trigger automatic `ocm` release
+  * **Override:** Skip root release if `--suppress-root-release` flag is set (for coordinated releases)
+* Update `/VERSIONS.yaml` via OCM bot PR with new sub-component version and computed root version (if applicable)
+* Failure handling: if VERSIONS.yaml update conflicts, stop pipeline and alert maintainers
+
+**Manual Parameters:**
+
+* `suppress-root-release`: Boolean flag to prevent automatic OCM root release (for coordinated multi-component releases as described [above](#edge-case-coordinated-multi-component-releases))
+* `release_candidate`: Boolean flag to create a release candidate tag (e.g., `v0.30.0-rc.1`) instead of a final release
+* `release_candidate_name`: Release candidate suffix (e.g., `rc.1`, `rc.2`) when `release_candidate` is true
+
```

Comment:
The less parameters the less failure prone this whole thing will be. I don't want a typo to kill the entire release process.